streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -105,6 +105,7 @@
 
     private Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> standbyRecords;
     private boolean processStandbyRecords = false;
+    private AtomicBoolean stateStoresAvailable = new AtomicBoolean(false);
```

Comment:
nit: could we rename this boolean to `initialized`? And then when checking this flag is not true:

```
throw new InvalidStateStoreException("Store: " + storeName + " may not currently available, since the hosted thread is not (re-)initialized yet");
```

Also we may need to update the `ReadOnlyKeyValueStore` and `ReadOnlyWindowStore` about those possible thrown exceptions and how to handle them (e.g. backoff and retry).